src/stdlib_experimental_io.f90

```diff
@@ -1,12 +1,31 @@
 module stdlib_experimental_io
-use iso_fortran_env, only: dp=>real64
+use iso_fortran_env, only: sp=>real32, dp=>real64
 implicit none
 private
 public :: loadtxt, savetxt
 
+interface loadtxt
+    module procedure sloadtxt
+    module procedure dloadtxt
+end interface
+
+interface savetxt
+    module procedure ssavetxt
+    module procedure dsavetxt
+end interface
+
 contains
 
-subroutine loadtxt(filename, d)
+subroutine sloadtxt(filename, d)
+character(len=*), intent(in) :: filename
+real(sp), allocatable, intent(out) :: d(:,:)
+real(dp), allocatable :: tmp(:,:)
+call dloadtxt(filename, tmp)
```

Comment:
*This implies a additional copy of the array d (in dp). This could be quite inefficient for large files/arrays.
*This way could be also difficult to generalize for quad precision qp.

I implemented a more general solution and extended it to qp. How could I propose these changes?